movement/movement.c

```diff
@@ -76,6 +82,20 @@
 movement_state_t movement_state;
 void * watch_face_contexts[MOVEMENT_NUM_FACES];
 watch_date_time scheduled_tasks[MOVEMENT_NUM_FACES];
+
+// HPT stuff
+
+// bit flags indicating which watch face wants the HPT enabled.
+// TODO make sure this number width is larger than the number of faces
+#define HPT_REQUESTS_T uint16_t
+HPT_REQUESTS_T hpt_enable_requests;
```

Comment:
How about this?

```C
#define HPT_REQUESTS_SIZE (MOVEMENT_NUM_FACES / CHAR_BIT)

unsigned char hpt_enable_requests[HPT_REQUESTS_SIZE + 1];
```

Reply:
I only really need boolean flags, and I was trying to come up with a good way to pack them into a smaller structure to save memory.  (On the other hand, I went and allocated 64-bits for a bunch of clock face timestamps that probably aren't going to use them)

Reply:
Oh I see what you're doing with that. That would also work, and I think once I get this feature working, I'll do another pass and try to optimize memory usage as much as possible. (like, timestamps really only need to be 40-bits, even in 1024hz mode, but packing all that stuff up is a lot of extra math)

Reply:
With some extra preprocessor hackery, we could get rid of that `+1` as well which is there just to prevent zero sized array errors in case the result is zero. That would save an extra byte.